### DISCUSSION TITLE
How to register Tauri app as a browser?

### DISCUSSION THREAD

**Discussion by shinebayar-g** [tauri-apps/tauri#9184] [Q&A] [answered]
Hi. I'm trying to build a similar app to https://github.com/johnste/finicky. My app should be able to register itself as a browser. Is it possible to do so using Tauri?

Related https://developer.apple.com/documentation/xcode/preparing-your-app-to-be-the-default-browser

**Comment by FabianLars** [comment, marked as answer]
The way finicky works is by using a combination of `CFBundleURLTypes` in Info.plist ([example](https://github.com/FabianLars/tauri-plugin-deep-link/blob/main/example/Info.plist)) and https://developer.apple.com/documentation/coreservices/1447760-lssetdefaulthandlerforurlscheme ([use in finicky](https://github.com/johnste/finicky/blob/33e12489fb1cdc948afc097dad0b8f91f7c8ade9/Finicky/Finicky/AppDelegate.swift#L103)) to set your app as the default (note that this function is deprecated, i'm not sure what it was replaced by). The first part is supported in v1 via the plugin i linked as the example and in v2 by tauri itself (docs yet to be written). There's no api for LSSetDefaultHandlerForURLScheme though so you'll have to do that yourself, maybe this [super old crate](https://docs.rs/launch-services/latest/launch_services/fn.set_default_handle_for_url_scheme.html) still works 🤷

**Reply by shinebayar-g** [reply]
I see, thanks so much @FabianLars. Is [this](https://github.com/tauri-apps/tauri/pull/8680) the related PR in v2?

**Reply by FabianLars** [reply]
Yes, but if you only care about macOS this PR wasn't even necessary since you can modify the Info.plist file yourself easily.
